### ISSUE TITLE
CustomLayers not registered in dimension until save occurs

### ISSUE THREAD

**Issue by IR0NSIGHT** [Captain-Chaos/WorldPainter#308]
Noticed this issue when looking through the code, am building a script that would like to use custom layers

if i add a new custom layer (used groundcover), it doesnt show up in the Dimension.customLayers list until its overwritten by App.saveCustomLayers().

a. bad for my script
b. dimension.getCustomLayer is used in 9 other classes... :D

easy proof:
run script, add couple custom layers, run script (layers dont show up), save, run script (layers do show up)
run as WP script from a .js file (cant upload, github doesnt allow)
```js
print("Default layers:");
var allLayers = dimension.getAllLayers(false);
allLayers =
  org.pepsoft.worldpainter.layers.LayerManager.getInstance().getLayers();
allLayers.forEach(function (element) {
  print("\t" + element.getName());
});
print("CUSTOM:");
allLayers = dimension.getCustomLayers();
allLayers.forEach(function (element) {
  print("\t" + element.getName() + " id: " + element.getId());
});
```

**Comment by Captain-Chaos** [comment]
Yes, that's by design. There's a number of entities that are only stored in the Dimension for cold storage. What are you trying to do for which you need access to all current custom layers?

**Comment by Captain-Chaos** [comment]
One direction you might investigate is trying to obtain the layers via the `App` class. That is the class that maintains the live list of `CustomLayer`s while a dimension is loaded, via an instance of `PaletteManager`. You can obtain it via `App.getInstanceIfExists()` and it has a 'getCustomLayers()` method.

When running the script via `wpscript` the `App` class doesn't exist, so you would have to detect that and handle it appropriately.

Would that be sufficient? I'm finding that this is too niche to justify putting a lot of time in, unfortunately, so I hope you can work around the issue in some manner.
